src/legacy/index.ts

```diff
@@ -86,12 +93,18 @@ async function buildGraph(
 
     depNodesIds.push(depNodeId);
 
-    builder.addPkgNode(depPkg, depNodeId);
+    const nodeInfo: types.NodeInfo = {};
+
+    if (dep.versionProvenance) {
+      nodeInfo.versionProvenance = dep.versionProvenance;
```

Comment:
should we be worried about hanging on to a reference here? potential source of bugs...

Reply:
not as far as I can see.. do you see something else?